File: domain/src/main/java/no/nav/melosys/domain/brev/Saksbehandlingstid.java
package no.nav.melosys.domain.brev;

import java.time.Instant;
import java.time.Period;

public class Saksbehandlingstid {

    public static final int SAKSBEHANDLINGSTID_UKER = 12;

    private Saksbehandlingstid() {
    }

    public static Instant beregnSaksbehandlingsfrist(Instant forsendelseMottatt) {
        return forsendelseMottatt.plus(Period.ofWeeks(SAKSBEHANDLINGSTID_UKER));
    }
}

File: domain/src/main/kotlin/no/nav/melosys/domain/brev/Saksbehandlingstid.kt
package no.nav.melosys.domain.brev

import java.time.Instant
import java.time.Period

object Saksbehandlingstid {
    const val SAKSBEHANDLINGSTID_UKER: Int = 12

    fun beregnSaksbehandlingsfrist(forsendelseMottatt: Instant?): Instant =
        forsendelseMottatt?.plus(Period.ofWeeks(SAKSBEHANDLINGSTID_UKER)) ?: error(
            "Forsendelse mottatt dato kan ikke være null når saksbehandlingstid skal beregnes"
        )
}
